Lib/zipfile/__init__.py

```diff
@@ -508,6 +518,24 @@ def _decodeExtra(self):
                 except struct.error:
                     raise BadZipFile(f"Corrupt zip64 extra field. "
                                      f"{field} not found.") from None
+            elif tp == 0x7075:
+                data = extra[4:ln+4]
+                # Unicode Path Extra Field
+                try:
+                    up_version, up_name_crc = unpack('<BL', data[:5])
+                    if up_version == 1 and up_name_crc == self.orig_filename_crc:
+                        up_unicode_name = data[5:].decode('utf-8')
+                        if up_unicode_name:
+                            self.filename = _sanitize_filename(up_unicode_name)
+                        else:
+                            import warnings
+                            warnings.warn("Empty unicode path extra field (0x7075)", stacklevel=2)
+                except struct.error:
+                    import warnings
+                    warnings.warn("Corrupt unicode path extra field (0x7075)", stacklevel=2)
+                except UnicodeDecodeError:
+                    import warnings
+                    warnings.warn('Corrupt unicode path extra field (0x7075): invalid utf-8 bytes', stacklevel=2)
```

Comment:
Please raise BadZipFile to follow other conventions.

Reply:
I modified it in a way that throws an exception, but given that this is an `extra` field, wouldn't it be okay to ignore it and let the 'original' filename field be used if an exception is thrown?